File: frontend/src/app/shared/service/collaborator/common/NewsCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {NewsDto} from '../../../model/common/News.model';
import {NewsCriteria} from '../../../criteria/common/NewsCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class NewsCollaboratorService extends AbstractService<NewsDto, NewsCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/news/';
    }

    public constrcutDto(): NewsDto {
        return new NewsDto();
    }

    public constrcutCriteria(): NewsCriteria {
        return new NewsCriteria();
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/common/NewsCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.common;

import java.util.List;
import ma.zs.alc.bean.core.common.News;
import ma.zs.alc.dao.criteria.core.common.NewsCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface NewsCollaboratorService extends  IService<News,NewsCriteria>  {

    List<News> findByCollaboratorId(Long id);
    int deleteByCollaboratorId(Long id);
    long countByCollaboratorId(Long id);




}